extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorTest.java

```diff
@@ -3668,6 +3840,59 @@ public TestableKafkaSupervisor(
           spec,
           rowIngestionMetersFactory
       );
+      this.consumerProperties = spec.getIoConfig().getConsumerProperties();
+    }
+
+    @Override
+    protected RecordSupplier<Integer, Long> setupRecordSupplier()
+    {
+      final Map<String, Object> consumerConfigs = new HashMap<>();
+      consumerConfigs.put("metadata.max.age.ms", "1");
+      consumerConfigs.put("group.id", StringUtils.format("kafka-supervisor-%s", RandomIdUtils.getRandomId()));
+      consumerConfigs.put("auto.offset.reset", "none");
+      consumerConfigs.put("enable.auto.commit", "false");
+      consumerConfigs.put("isolation.level", "read_committed");
+      final Properties props = new Properties();
+      consumerProperties.get("key.deserializer");
+      System.out.println("consumerProperties");
+      consumerProperties.forEach((key, value) -> System.out.println(key + ":" + value));
+      KafkaRecordSupplier.addConsumerPropertiesFromConfig(props, sortingMapper, consumerProperties);
+      props.putAll(consumerConfigs);
+      Deserializer keyDeserializerObject = getKafkaDeserializer(props, "key.deserializer");
+      Deserializer valueDeserializerObject = getKafkaDeserializer(props, "value.deserializer");
```

Comment:
You can simplify the test code by directly instantiating the deserializer here:
```java
      Deserializer keyDeserializerObject = new ByteArrayDeserializer();
      Deserializer valueDeserializerObject = new ByteArrayDeserializer();
```

and then delete the `getKafkaDeserializer()` method you added.